reana_db/models.py

```diff
@@ -435,6 +456,41 @@ def can_transition_to(self, next_status):
         return current_transition in ALLOWED_WORKFLOW_STATUS_TRANSITIONS
 
 
+@event.listens_for(Workflow.status, "set")
+def workflow_status_change_listener(workflow, new_status, old_status, initiator):
+    """Workflow status change listener."""
+    if new_status in [
+        WorkflowStatus.finished,
+        WorkflowStatus.failed,
+    ]:
+        workflow.run_finished_at = datetime.now()
+    elif new_status in [WorkflowStatus.stopped]:
+        workflow.run_stopped_at = datetime.now()
+
+    finished_at = workflow.run_finished_at or workflow.run_stopped_at
+    if workflow.run_started_at and finished_at:
+        RESOURCE = "cpu"
```

Comment:
Following the discussion during stand-up. We need to find a logic way of knowing which exact CPU resource a workflow has been consumed.

For this first implementation, we have taken the decision to calculate the CPU time at the workflow level. This includes:
- Container related overhead:
    - Meeting pod requirements:
       - Pulling images
       - Mounting volumes
       - Mounting config maps
       - etc...
    - Start pod
    - Note: it doesn't include containers teardown
- Workflow engine orchestration overhead
- The sum of all execution times for jobs

Although fine-grained accounting would be very beneficial, we cannot exactly account for each detail yet. Instead, we will have to account for all of it together. 

My suggestion would be to have a "general CPU" (I know the name isn't great, just coming up with one to kick-off the discussion) which would map one to one to the summary shown in the [your profile page](https://github.com/reanahub/reana-ui/pull/134), being a summary of total CPU/processing power usage. Once we get more advanced into what was consumed on each part, this UI wouldn't be enough and we would could have a dedicated page with a breakdown in general and per workflow.

Note: with the current implementation only `ResourceType.cpu` will be aggregated. If this "general CPU" would include GPUs too (which I think it makes sense) maybe the name has to be "more general" and the implementation should be amended.

Reply:
Maybe "default CPU"?
I guess for now all the workflows can consume this resource. I wonder if we should have a flag to mark a resource of given `ResourceType` as the default? So we avoid the hardcoding... But still, we'd have the problem of knowing if it was run by CPU or GPU...

Reply:
It makes totally sense to calculate this "general CPU". I guess I would call it with something englobing, like "Processor" or better "Processing" or even more completely "Processing time".  

Some of the use cases mention are perfectly normal to get accounted to the user. If User A has large Docker images, REANA needs to pull them, so it is an attribute of the user workflow... Unlike User B that only needs small images.  (If an image is already pulled, that's a sort of "luck" which can be neglected in the zero-order approach.) Ditto for orchestration time: if a User A has pipeline with 10000 parallel steps, and User B with only 100, it is also perfectly normal to count more for user A, since orchestration of these jobs takes time... Another attribute of the user workflow.
Ditto for mounting volumes...  So I wouldn't go into too much granularity here.

Reply:
Actions discussed after stand-up:
- [ ] Load fixtures with base `Resource`s (similar to the [current bash script](https://github.com/reanahub/reana/pull/416) but more user-friendly, technical detail to be decided)
- [ ] Create entries in `UserResource` when users are created (note that users can be created through UI and CLI). Whether we set default quota to `null` or `0`, to be dediced.

Still to be agreed upon:

> I wonder if we should have a flag to mark a resource of given ResourceType as the default? So we avoid the hardcoding...

This makes a lot of sense. I see two ways of achieving this:
1. Static config: Treating this as "before deployment" static configuration that would be populated when we ingest the fixtures by selecting one of the `Resource`s as default (see https://github.com/reanahub/reana/issues/277). This would technically require a restart of all the pods that read/require the configuration.
2. Dynamic config: create a new table `Configuration` which would look a bit like follows (very early thought, take with a grain of salt):

    | Scope | Name |  Value | Default | Description | Optional |
    | ------ | ------ | ------ | ------- | ------------ | -------- |
    | global | resource:processing_time:default | null | d717a177-6e99-43d9-b978-d35d11e53048 (FK Resource.id_) | Processing time resource to account for when there is no specific resource to assign to. | false |

    This would scale towards the idea of having [admin roles](https://github.com/reanahub/reana-server/issues/285) changing this configuration through the UI. I've added `scope` as the concept that the default processing time resource might changeable in the future by groups admins (admin of: a collaboration, a working group etc..)]

Reply:
Do we need a new table? All the resources-under-quota are basically defined in a table already, so why not to reuse that one? E.g. title/type/description etc is already there... Whether REANA instances will populate it by fixtures, or "by hand", shouldn't matter.

The admins adding new resources by a web UI is perhaps looking too much ahead; they would need to modify source code to make these resources registered as needed, which is a complex task... What I thought about as admin UI for MVP are easily editable things, i.e. *keeping* the IDs and names (that we use as a sort of ID) and altering only presentation stuff like title and description, or altering default quota values, etc.... But no adding/deleting rows, as that would require to change the code base to start collecting those new resource values.

Reply:
I see no way, with the current implementation, to solve the problem of choosing a specific row in the `Resource` table to account for a given workflow run. Currently, this is hardcoded.

I thought that a way to tackle the problem is to store somewhere a pointer to the row that represents "default resource for processing time" for this specific case, and in general, would be a pointer to default resources (be it GPU, disk etc...). In my eyes, this pointer could be in a config file or in a database table.

Reply:
Idea: What if we have two new columns in the `Resource` table called `resource_unit` (yes, I know, it's an awful name but I can't find anything better atm) that can be either `ResourceUnitType.processing` or `ResourceUnitType.storage` and `default` (boolean), which represents if that specific resource unit is the default one. Then we can have a constraint that forces that there can only be a unique default=True per `ResourceUnitType`. Would that work? Maybe it's a bit crazy... bear with me